Mathlib/Analysis/Convex/SpecificFunctions/Basic.lean

```diff
@@ -147,7 +144,47 @@ theorem one_add_mul_self_le_rpow_one_add {s : ℝ} (hs : -1 ≤ s) {p : ℝ} (hp
   exact (one_add_mul_self_lt_rpow_one_add hs hs' hp).le
 #align one_add_mul_self_le_rpow_one_add one_add_mul_self_le_rpow_one_add
 
-/- For `p : ℝ` with `1 < p`, `fun x ↦ x ^ p` is strictly convex on $[0, +∞)$.
+/-- **Bernoulli's inequality** for real exponents, strict version: for `0 < p < 1` and `-1 ≤ s`,
+with `s ≠ 0`, we have `(1 + s) ^ p < 1 + p * s`. -/
+theorem rpow_one_add_lt_one_add_mul_self {s : ℝ} (hs : -1 ≤ s) (hs' : s ≠ 0) {p : ℝ} (hp1 : 0 < p)
+    (hp2 : p < 1) : (1 + s) ^ p < 1 + p * s := by
+  rcases eq_or_lt_of_le hs with (rfl | hs)
+  · have : p ≠ 0 := by positivity
+    simpa [zero_rpow this]
+  have hs1 : 0 < 1 + s := by linarith
+  rcases le_or_lt (1 + p * s) 0 with hs2 | hs2
+  · nlinarith
+  rw [rpow_def_of_pos hs1, ← exp_log hs2]
+  apply exp_strictMono
+  have hs3 : 1 + s ≠ 1 := by contrapose! hs'; linarith
+  have hs4 : 1 + p * s ≠ 1 := by contrapose! hs'; nlinarith
+  cases' lt_or_gt_of_ne hs' with hs' hs'
+  · rw [← lt_div_iff hp1, ← div_lt_div_right_of_neg hs']
+    haveI : (1 : ℝ) ∈ Ioi 0 := zero_lt_one
+    convert strictConcaveOn_log_Ioi.secant_strict_mono this hs1 hs2 hs3 hs4 _ using 1
+    · field_simp
+    · field_simp
+    · nlinarith
+  · rw [← lt_div_iff hp1, ← div_lt_div_right hs']
+    haveI : (1 : ℝ) ∈ Ioi 0 := zero_lt_one
+    convert strictConcaveOn_log_Ioi.secant_strict_mono this hs2 hs1 hs4 hs3 _ using 1
+    · field_simp
+    · field_simp
+    · nlinarith
```

Comment:
This takes 3.2 seconds to run on my machine, which is an eternity for a relatively simple statement like this one. The vast majority of that is coming from the heavyweight tactics `nlinarith` and `field_simp` which you use four times each. 

Most – possibly all – of these can be replaced with slightly uglier, but hugely quicker, longhand arguments with `rw` etc.

Reply:
On further inspection, I see now that you are quite closely following the existing proof in the `1 ≤ p` case, which is similarly cavalier in its use of multiple heavyweight tactics. But the fact that slow code exists in the library isn't a justification for adding more bad code!

Here's a version which is structurally very similar to what you wrote, but takes something like one-sixth of the time to compile:
````
/-- **Bernoulli's inequality** for real exponents, strict version: for `0 < p < 1` and `-1 ≤ s`,
with `s ≠ 0`, we have `(1 + s) ^ p < 1 + p * s`. -/
theorem rpow_one_add_lt_one_add_mul_self {s : ℝ} (hs : -1 ≤ s) (hs' : s ≠ 0) {p : ℝ} (hp1 : 0 < p)
    (hp2 : p < 1) : (1 + s) ^ p < 1 + p * s := by
  rcases eq_or_lt_of_le hs with (rfl | hs)
  · simpa only [add_right_neg, zero_rpow hp1.ne', mul_neg, mul_one, lt_add_neg_iff_add_lt,
      zero_add]
  have hs1 : 0 < 1 + s := neg_lt_iff_pos_add'.mp hs
  have hs2 : 0 < 1 + p * s := by 
    rw [← neg_lt_iff_pos_add']
    rcases lt_or_gt_of_ne hs' with h | h
    · exact hs.trans (lt_mul_of_lt_one_left h hp2)
    · exact neg_one_lt_zero.trans (mul_pos hp1 h)
  have hs3 : 1 + s ≠ 1 := hs' ∘ add_right_eq_self.mp
  have hs4 : 1 + p * s ≠ 1 := by 
    contrapose! hs'; rwa [add_right_eq_self, mul_eq_zero, eq_false_intro hp1.ne', false_or] at hs'
  rw [rpow_def_of_pos hs1, ← exp_log hs2]
  apply exp_strictMono
  cases' lt_or_gt_of_ne hs' with hs' hs'
  · rw [← lt_div_iff hp1, ← div_lt_div_right_of_neg hs']
    convert strictConcaveOn_log_Ioi.secant_strict_mono (zero_lt_one' ℝ) hs1 hs2 hs3 hs4 _ using 1
    · rw [add_sub_cancel', div_div, log_one, sub_zero]
    · rw [add_sub_cancel', log_one, sub_zero]
    · apply add_lt_add_left (lt_mul_of_lt_one_left hs' hp2)
  · rw [← lt_div_iff hp1, ← div_lt_div_right hs']
    convert strictConcaveOn_log_Ioi.secant_strict_mono (zero_lt_one' ℝ) hs2 hs1 hs4 hs3 _ using 1
    · rw [add_sub_cancel', log_one, sub_zero]
    · rw [add_sub_cancel', div_div, log_one, sub_zero]
    · apply add_lt_add_left (mul_lt_of_lt_one_left hs' hp2)
````

I suggest you have a go at reworking the other proofs similarly. E.g. it should be possible to get `strictConcaveOn_rpow` (currently 5.8s on my machine) down to something more civilised.

Reply:
Done.